PlotJuggler does not build with CLang

When building PlotJuggler with CLang, the following error occurs:
```
.../PlotJuggler/plugins/ROS/RosMsgParsers/ros_parser.cpp:123:52: error: 
      non-constant-expression cannot be narrowed from type 'size_t' (aka 'unsigned long') to
      'nonstd::span_lite::span::index_type' (aka 'long') in initializer list [-Wc++11-narrowing]
                {const_cast<uint8_t*>(msg.data()), msg.size()},
                                                   ^~~~~~~~~~
.../PlotJuggler/plugins/ROS/RosMsgParsers/ros_parser.cpp:123:52: note: insert an
      explicit cast to silence this issue
                {const_cast<uint8_t*>(msg.data()), msg.size()},
                                                   ^~~~~~~~~~
                                                   static_cast<index_type>( )
```
On GCC, this is only a warning. CLang seems to be more restrictive here.

--- a/plugins/ROS/RosMsgParsers/ros_parser.cpp
+++ b/plugins/ROS/RosMsgParsers/ros_parser.cpp
@@ -120,7 +120,7 @@ void RosMessageParser::pushMessageRef(const std::string &topic_name,
 
     bool max_size_ok = _introspection_parser->deserializeIntoFlatContainer(
                 topic_name,
-                {const_cast<uint8_t*>(msg.data()), msg.size()},
+                {const_cast<uint8_t*>(msg.data()), static_cast<long>(msg.size())},
                 &flat_container,
                 _max_array_size );
 

